pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -81,6 +83,33 @@ func writeVifFile(buf []byte, pid, name string) error {
 	return nil
 }
 
+func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid string) error {
+	link, err := Handler.LinkByName(podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// get IP address
+	addrList, err := Handler.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// no ip assigned. ipam disabled
+	if len(addrList) == 0 {
+		return nil
+	}
+
+	err = writeToCachedFile(PodCacheInterface{Iface: iface, PodIP: addrList[0].IP.String()},
```

Comment:
This file needs to be deleted.

My suggestion is to delete the entire UID directory for a vm during the processVMCleanup() function in vm.go.

Reply:
Added now